app/models/cloud_volume.rb

```diff
@@ -12,4 +15,60 @@ def self.available
     joins("LEFT OUTER JOIN disks ON disks.backing_id = cloud_volumes.id")
       .where("disks.backing_id" => nil)
   end
+
+  def self.class_by_ems(ext_management_system)
+    # TODO(lsmola) taken from OrchesTration stacks, correct approach should be to have a factory on ExtManagementSystem
+    # side, that would return correct class for each provider
+    ext_management_system.class::CloudVolume
+  end
+
+  def self.create_volume(ext_management_system, options = {})
+    klass = class_by_ems(ext_management_system)
+    tenant = options[:cloud_tenant]
+
+    created_volume = klass.raw_create_volume(ext_management_system, options)
+
+    klass.create(
+      :name                  => options[:display_name],
+      :ems_ref               => created_volume[:ems_ref],
+      :status                => created_volume[:status],
+      :size                  => options[:size].to_i.gigabytes,
+      :ext_management_system => ext_management_system,
+      :cloud_tenant          => tenant)
+  end
+
+  def self.validate_create_volume(ext_management_system)
+    klass = class_by_ems(ext_management_system)
+    return klass.validate_create_volume(ext_management_system) if ext_management_system &&
+                                                                  klass.respond_to?(:validate_create_volume)
+    validate_unsupported("Create Volume Operation")
+  end
+
+  def self.raw_create_volume(_ext_management_system, _options = {})
+    raise NotImplementedError, "raw_create_volume must be implemented in a subclass"
```

Comment:
Hrm, instead of `NotImplemented` here, maybe `validated_unsupported`?  @Fryguy is that right?